optimade/server/routers/landing.py

```diff
@@ -1,39 +1,88 @@
-""" OPTIMADE landing page, rendered as a Jinja2 template. """
+""" OPTIMADE landing page router. """
 
 from pathlib import Path
+from functools import lru_cache
 
 from fastapi import Request
-from fastapi.templating import Jinja2Templates
+from fastapi.responses import HTMLResponse
 from starlette.routing import Router, Route
 from optimade import __api_version__
 
 from optimade.server.routers import ENTRY_COLLECTIONS
 from optimade.server.routers.utils import meta_values, get_base_url
 from optimade.server.config import CONFIG
 
-template_dir = Path(__file__).parent.joinpath("static").resolve()
-TEMPLATES = Jinja2Templates(directory=[template_dir])
 
+@lru_cache()
+def render_landing_page(url: str) -> HTMLResponse:
+    """Render and cache the landing page.
 
-async def landing(request: Request):
-    """Show a human-readable landing page when the base URL is accessed."""
+    This function uses the template file `./static/landing_page.html`, adapted
+    from the original Jinja template. Instead of Jinja, some basic string
+    replacement is used to fill out the fields from the server configuration.
 
-    meta = meta_values(request.url, 1, 1, more_data_available=False)
+    !! warning
+    The removal of Jinja means that the fields are no longer validated as
+    web safe before inclusion in the template.
+
+    """
+    meta = meta_values(url, 1, 1, more_data_available=False)
     major_version = __api_version__.split(".")[0]
-    versioned_url = f"{get_base_url(request.url)}/v{major_version}/"
+    versioned_url = f"{get_base_url(url)}/v{major_version}/"
+
+    template_dir = Path(__file__).parent.joinpath("static").resolve()
 
-    context = {
-        "request": request,
-        "request_url": request.url,
+    with open(template_dir / "landing_page.html", "r") as f:
+        html = "\n".join(f.readlines())
+
+    # Build a dictionary that maps the old Jinja keys to the new simplified replacements
+    replacements: dict[str, str] = {
```

Comment:
Shouldn't this be `Dict[str, str]` instead, where `Dict` is imported from `typing`?

Reply:
I guess so, `Dict` is deprecated in >=3.9 but I'm surprised this didn't fail for 3.8. Tbh its not a particularly important type hint so I'll just remove it.


```suggestion
    replacements = {
```

Reply:
Really? Huh. Ah yes, I've just seen this in the documentation. Very good. Just keep it in I suppose.